test/e2e/certificate_fastdns.go

```diff
@@ -0,0 +1,123 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+
+	api "github.com/appscode/voyager/apis/voyager/v1beta1"
+	"github.com/appscode/voyager/pkg/certificate"
+	"github.com/appscode/voyager/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("CertificateWithDNSProviderFastDNS", func() {
+	var (
+		f                *framework.Invocation
+		cert             *api.Certificate
+		userSecret       *core.Secret
+		credentialSecret *core.Secret
+	)
+
+	BeforeEach(func() {
+		f = root.Invoke()
+
+		skipTestIfFastdnsSecretNotProvided()
+		if !options.TestCertificate {
+			Skip("FastDNS Certificate Test is not enabled")
+		}
+
+		userSecret = &core.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-" + f.Certificate.UniqueName(),
+				Namespace: f.Namespace(),
+			},
+			Data: map[string][]byte{
+				api.ACMEUserEmail: []byte("sadlil@appscode.com"),
+				api.ACMEServerURL: []byte(certificate.LetsEncryptStagingURL),
+			},
+		}
+
+		_, err := f.KubeClient.CoreV1().Secrets(userSecret.Namespace).Create(userSecret)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	BeforeEach(func() {
+		f = root.Invoke()
+
+		fmt.Println("TEST_AKAMAI_DNS_DOMAINS", os.Getenv("TEST_AKAMAI_DNS_DOMAINS"))
+
+		credentialSecret = &core.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cred-" + f.Certificate.UniqueName(),
+				Namespace: f.Namespace(),
+			},
+			Data: map[string][]byte{
+				"AKAMAI_HOST":          []byte(os.Getenv("TEST_AKAMAI_HOST")),
+				"AKAMAI_CLIENT_TOKEN":  []byte(os.Getenv("TEST_AKAMAI_CLIENT_TOKEN")),
+				"AKAMAI_CLIENT_SECRET": []byte(os.Getenv("TEST_AKAMAI_CLIENT_SECRET")),
+				"AKAMAI_ACCESS_TOKEN":  []byte(os.Getenv("TEST_AKAMAI_ACCESS_TOKEN")),
+			},
+		}
+
+		_, err := f.KubeClient.CoreV1().Secrets(credentialSecret.Namespace).Create(credentialSecret)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	BeforeEach(func() {
+		cert = f.Certificate.GetSkeleton()
+		cert.Spec = api.CertificateSpec{
+			Domains: []string{os.Getenv("TEST_AKAMAI_DNS_DOMAINS")},
+			ChallengeProvider: api.ChallengeProvider{
+				DNS: &api.DNSChallengeProvider{
+					Provider:             "fastdns",
+					CredentialSecretName: credentialSecret.Name,
+				},
+			},
+			ACMEUserSecretName: userSecret.Name,
+			Storage: api.CertificateStorage{
+				Secret: &core.LocalObjectReference{},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Creating certificate with" + cert.Name)
+		err := f.Certificate.Create(cert)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if options.Cleanup {
```

Comment:
Cleanup certificate object
```
f.Certificate.Delete(cert)
```

Reply:
Good catch @diptadas! It is fixed now. Thanks!